kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseInstanceMigrationIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+        KieServerDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/case-insurance-101").getFile());
+    }
+
+    @Before
+    public void cleanContainers() {
+        disposeAllContainers();
+        // use different aliases to avoid policy based removal - keep latest only
+        createContainer(CONTAINER_ID, releaseId, CONTAINER_ID);
+        createContainer(CONTAINER_ID_2, releaseId101, CONTAINER_ID_2);
+    }
+    
+    
+    @Test
+    public void testCarInsuranceClaimCaseMigration() throws Exception {
+        // start case with users assigned to roles
+        String caseId = startCarInsuranceClaimCase(USER_YODA, USER_JOHN);
+        // let's verify case is created
+        assertCaseInstance(caseId);
+        
+        Map<String, String> processMapping = new HashMap<>();
+        processMapping.put(CLAIM_CASE_DEF_ID, CLAIM_CASE_DEF_ID_2);
+        
+        CaseMigrationReportInstance report = caseAdminClient.migrateCaseInstance(CONTAINER_ID, caseId, CONTAINER_ID_2, processMapping);
+        assertThat(report).isNotNull();
+        assertThat(report.isSuccessful()).isTrue();
+        
+        assertMigratedCaseInstance(caseId);
+    }
+    
+    @Test
+    public void testCarInsuranceClaimCaseMigrationWithSubprocess() throws Exception {
+        // start case with users assigned to roles
+        String caseId = startCarInsuranceClaimCase(USER_YODA, USER_JOHN);
+        // let's verify case is created
+        assertCaseInstance(caseId);
+                
+        caseClient.addDynamicSubProcess(CONTAINER_ID, caseId, PROCESS_DEF_ID, null);
+        
+        List<ProcessInstance> processInstances = caseClient.getProcessInstances(CONTAINER_ID, caseId, Arrays.asList(1), 0, 10);
+        assertThat(processInstances).isNotNull();
+        assertThat(processInstances).hasSize(2);
+        
+        for (ProcessInstance pi : processInstances) {
+            assertThat(pi.getContainerId()).isEqualTo(CONTAINER_ID);   
+        }
+        
+        Map<String, String> processMapping = new HashMap<>();
+        processMapping.put(CLAIM_CASE_DEF_ID, CLAIM_CASE_DEF_ID_2);
+        processMapping.put(PROCESS_DEF_ID, PROCESS_DEF_ID_2);
+        
+        CaseMigrationReportInstance report = caseAdminClient.migrateCaseInstance(CONTAINER_ID, caseId, CONTAINER_ID_2, processMapping);
+        assertThat(report).isNotNull();
+        assertThat(report.isSuccessful()).isTrue();
+        
+        assertMigratedCaseInstance(caseId);
+        
+        processInstances = caseClient.getProcessInstances(CONTAINER_ID_2, caseId, Arrays.asList(1), 0, 10);
+        assertThat(processInstances).isNotNull();
+        assertThat(processInstances).hasSize(2);
+        
+        for (ProcessInstance pi : processInstances) {
+            assertThat(pi.getContainerId()).isEqualTo(CONTAINER_ID_2);   
+        }
+    }
+
+    
```

Comment:
Does it make sense to include nodeMapping test and one error test?

Reply:
be my guest ;) in other words would be good that someone else tries the api to spot any issues

Reply:
that's exactly what I thought about :) Will include these test perhaps some time this or next week.